spaghetti-typescript-support/src/main/groovy/com/prezi/spaghetti/typescript/TypeScriptHeaderGenerator.groovy

```diff
@@ -32,14 +29,23 @@ class TypeScriptHeaderGenerator extends AbstractHeaderGenerator {
 		def header = params.header
 		copySpaghettiClass(outputDirectory)
 		generateLocalModule(config.localModule, outputDirectory, header)
-		config.directDependentModules.each { dependentModule ->
+		(config.directDependentModules + config.lazyDependentModules).each { dependentModule ->
 			generateDependentModule(dependentModule, outputDirectory, header, true)
 		}
+		config.lazyDependentModules.each { dependentModule ->
+			generateLazyGetter(dependentModule, outputDirectory)
+		}
 		config.transitiveDependentModules.each { dependentModule ->
 			generateDependentModule(dependentModule, outputDirectory, header, false)
 		}
 	}
 
+	private static generateLazyGetter(EntityWithModuleMetaData<ModuleNode> module, File outputDirectory) {
+		def file = new File(outputDirectory, module.entity.name.replace(".", "_") + "__loader_generated.d.ts")
+		def accessorName = GeneratorUtils.createLazyModuleAccessorName(module.entity.name)
+		file << """declare function ${accessorName}(): Promise<${module.entity.name}.LazyModule>;"""
```

Comment:
If the return type was `Promise<typeof ${module.entity.name}>` then any module could be loaded async without needing a refactor to implement `LazyModule`.